src/Elliptic/DiscontinuousGalerkin/NumericalFluxes/FirstOrderInternalPenalty.hpp

```diff
@@ -0,0 +1,388 @@
+      normal_dot_flux(
+          make_not_null(
+              &get<NormalDotDivFlux<auxiliary_field_tag>>(*packaged_data)),
+          interface_unit_normal,
+          get<::Tags::Flux<field_tag, tmpl::size_t<Dim>, Frame::Inertial>>(
+              principal_div_aux_field_fluxes));
+      normal_dot_flux(
+          make_not_null(&get<NormalDotNormalDotFlux<auxiliary_field_tag>>(
+              *packaged_data)),
+          interface_unit_normal,
+          get<::Tags::Flux<field_tag, tmpl::size_t<Dim>, Frame::Inertial>>(
+              principal_ndot_aux_field_fluxes));
+    };
+    EXPAND_PACK_LEFT_TO_RIGHT(helper(FieldTags{}, AuxiliaryFieldTags{},
+                                     normal_dot_auxiliary_field_fluxes));
+  }
+
+  // This function combines local and remote data to the numerical fluxes.
+  // The numerical fluxes as not-null pointers are the first arguments. The
+  // other arguments are the packaged types for the interior side followed by
+  // the packaged types for the exterior side.
+  void operator()(
+      const gsl::not_null<db::item_type<::Tags::NormalDotNumericalFlux<
+          FieldTags>>*>... numerical_flux_for_fields,
+      const gsl::not_null<db::item_type<::Tags::NormalDotNumericalFlux<
+          AuxiliaryFieldTags>>*>... numerical_flux_for_auxiliary_fields,
+      const db::item_type<::Tags::NormalDotFlux<
+          AuxiliaryFieldTags>>&... normal_dot_auxiliary_flux_interiors,
+      const db::item_type<NormalDotDivFlux<
+          AuxiliaryFieldTags>>&... normal_dot_div_auxiliary_flux_interiors,
+      const db::item_type<NormalDotNormalDotFlux<
+          AuxiliaryFieldTags>>&... ndot_ndot_aux_flux_interiors,
+      const db::item_type<::Tags::NormalDotFlux<
+          AuxiliaryFieldTags>>&... minus_normal_dot_auxiliary_flux_exteriors,
+      const db::item_type<NormalDotDivFlux<
+          AuxiliaryFieldTags>>&... minus_normal_dot_div_aux_flux_exteriors,
+      const db::item_type<NormalDotDivFlux<
+          AuxiliaryFieldTags>>&... ndot_ndot_aux_flux_exteriors) const
+      noexcept {
+    // Need polynomial degress and element size to compute this dynamically
+    const double penalty = penalty_parameter_;
+
+    const auto helper = [&penalty](
+        const auto numerical_flux_for_field,
+        const auto numerical_flux_for_auxiliary_field,
+        const auto normal_dot_auxiliary_flux_interior,
+        const auto normal_dot_div_auxiliary_flux_interior,
+        const auto ndot_ndot_aux_flux_interior,
+        const auto minus_normal_dot_auxiliary_flux_exterior,
+        const auto minus_normal_dot_div_aux_flux_exterior,
+        const auto ndot_ndot_aux_flux_exterior) noexcept {
+      for (auto it = numerical_flux_for_auxiliary_field->begin();
+           it != numerical_flux_for_auxiliary_field->end(); it++) {
+        const auto index =
+            numerical_flux_for_auxiliary_field->get_tensor_index(it);
+        *it = 0.5 * (normal_dot_auxiliary_flux_interior.get(index) -
```

Comment:
Could you please add comments clarifying the sign here? Is "minus_normal" minus the interior normal or exterior normal? (I think it has to be minus the interior normal for the signs to work out)